src/main/java/com/glencoesoftware/omero/zarr/ZarrPixelsService.java

```diff
@@ -325,7 +325,7 @@ protected ZarrPixelBuffer createOmeNgffPixelBuffer(Pixels pixels) {
             String uri = getUri(image);
             if (uri == null) {
                 // Quick exit if we think we're OME-NGFF but there is no URI
-                if ("OMEXML".equals(image.getFormat().getValue())) {
+                if (image.getFormat() != null && "OMEXML".equals(image.getFormat().getValue())) {
```

Comment:
Should probably just be:

`"OMEXML".equals(unwrap(image.getFormat()))`

With a static import of `omero.rtypes.unwrap`. Similar examples can be see in `com.glencoesoftware.omero.ms.image.region.ImageDataRequestHandler`.

Reply:
Unfortunately, this does not work out of the box since `image.getFormat()` returns an `omero.model.Format` object and I get

```
/Users/sbesson/Documents/GitHub/omero-zarr-pixel-buffer/src/main/java/com/glencoesoftware/omero/zarr/ZarrPixelsService.java:330: error: incompatible types: Format cannot be converted to RType
                if ("OMEXML".equals(unwrap(image.getFormat()))) {
```

Could look into using the `OriginalFileFormatEnumHandler` to find the proper enumeration and return the IObject but I wonder if it's overkill.

Reply:
Scratch what I said. Since it's an instance of `ome.model.Image` (server side object) what you have currently is all we can really do.